### PR TITLE
Adding slack attachment on messages without plugin (and mobile)

### PR DESCRIPTION
**Pull request by jespino**
The message shown when you don't have the plugin or you are in mobile is way
worse than the one with the plugin. This change fill a bit the gap using an
slack attachment that is more similar to the custom post format.

### CODE REVIEW

File: server/http.go [hunk 1]
```diff
@@ -269,13 +280,23 @@ type startMeetingRequest struct {
...
+		Fallback: fmt.Sprintf("Video Meeting started at [%d](%s).\n\n[Join Meeting](%s)", meetingID, meetingURL, meetingURL),
+		Title:    topic,
+		Text:     fmt.Sprintf("Personal Meeting ID (PMI) : [%d](%s)\n\n[:movie_camera:  Join Meeting](%s)", meetingID, meetingURL, meetingURL),
+	}
 
 	post := &model.Post{
 		UserId:    creator.Id,
 		ChannelId: channelID,
-		Message:   fmt.Sprintf("Meeting started at %s.", meetingURL),
+		Message:   fmt.Sprintf("%s has started a meeting", creator.Username),
```
Comment (larkox) [inline server/http.go:296]: Do you mind updating this message to be the same as the one of the custom post? (i.e. `I have started a meeting`)

File: server/http.go [hunk 2]
```diff
@@ -238,8 +238,19 @@ func (p *Plugin) handleMeetingEnded(w http.ResponseWriter, r *http.Request, webh
...
+	length := int((post.UpdateAt - post.CreateAt) / 60)
+	startText := start.Format("Mon Jan 2 15:04:05 -0700 MST 2006")
+
+	slackAttachment := model.SlackAttachment{
+		Fallback: fmt.Sprintf("Meeting has ended: started at %s, lenght: %d minute(s).", startText, length),
+		Title:    "Meeting Summary",
+		Text:     fmt.Sprintf("Date: %s\n\nMeeting Length: %d minute(s)", startText, length),
+	}
+
 	post.Message = "Meeting has ended."
```
Comment (larkox) [inline server/http.go:251]: Do you mind updating this text so it is the same as the custom type? (i.e. `I have ended the meeting`)

File: server/http.go (hunk 1 above)
Comment (jespino) [reply server/http.go]: Done

File: server/http.go (hunk 2 above)
Comment (jespino) [reply server/http.go]: Done

**Comment by larkox** [comment]
@asaadmahmood I leave you here some screenshots of this functionality. This will make the messages on mobile more similar to what is seen on desktop by the use of Slack Attachments. A nice side effect is that whenever the plugin is disabled, the messages from Zoom also look like this (more similar to the messages with zoom enabled). This should not happen often, but it is an improvement on how they look.

This is how the messages look when the plugin is enabled (no changes here):
- Meeting created
Screenshot from 2020-05-28 12-35-55
- Meeting ended
Screenshot from 2020-05-28 12-39-02

Those will transform to these messages when the plugin is disabled (mobile view will be almost the same):
- Meeting created
Screenshot from 2020-05-28 12-37-00
- Meeting disabled
Screenshot from 2020-05-28 12-39-14

Keep in mind that since the plugin is disabled / on mobile, we are limited to Slack attachments here.

**Comment by asaadmahmood** [comment]
@larkox We can still use the zoom icon instead of the black icon right? `icon-brand-zoom` is in the fonts.

**Comment by larkox** [comment]
@asaadmahmood As far as I know, we can only add markdown or emojis (that camera is an emoji). We cannot inject HTML.

**Comment by jespino** [comment]
@larkox @asaadmahmood maybe one option is add a custom emoji from the plugin at startup, search for the custom emoji `:zoom-meeting-icon:` and if is not there, add it (Is a bit hacky and needs custom emojis support enabled, and can be affected by the users or other plugins). Another option is to allow plugins to register its own emojis as part of the "system emojis".

**Comment by jespino** [comment]
@larkox seeing the examples that you show, I think the minutes are not calculated properly (probably I have to check that) and probably we can add the first paragraph available in the version with the plugin, to make it fully consistent.

**Comment by asaadmahmood** [comment]
How much effort is it to add support for our icon fonts in the interactive dialog?

**Comment by jespino** [comment]
@asaadmahmood this is an slack attachment, so that would imply to expose somehow the icon font to the markdown, and that open the posibility of everybody using that, probably @hmhealey can give you more information about hard is to add that. This would need support for webapp and mobile, and it would be backward incompatible, in the sense that this change for the plugin is not going to work in older versions of mattermost.

Said that, I think we can improve that part in the future, but the current changes is a big improvement over the existing implementation, and I think making the changes that you are proposing is going to take a lot of time.

By the way, if you just pick any other emoji available instead of the camera, or we remove the icon, it would be a matter of minutes.

**Comment by asaadmahmood** [comment]
@jespino Then I think we can just remove the icon for now.

**Comment by jespino** [comment]
@asaadmahmood perfect, I'll do it :)

File: server/http.go [hunk 3]
```diff
@@ -238,8 +238,28 @@ func (p *Plugin) handleMeetingEnded(w http.ResponseWriter, r *http.Request, webh
...
+	topic, ok := post.Props["meeting_topic"].(string)
+	if !ok {
+		topic = "Zoom Meeting"
+	}
+
+	slackAttachment := model.SlackAttachment{
+		Fallback: fmt.Sprintf("Meeting %s has ended: started at %s, lenght: %d minute(s).", post.Props["meeting_id"], startText, length),
+		Title:    topic,
+		Text: fmt.Sprintf(
+			"Personal Meeting ID (PMI) : %d\n\n##### Meeting Summary\n\nDate: %s\n\nMeeting Length: %d minute(s)",
```
Comment (larkox) [inline server/http.go:253]: @jespino Something feels wrong in this formatting. Take a look to the screenshot.
Screenshot from 2020-05-29 18-33-22

**Comment by asaadmahmood** [comment]
@larkox Looks good to me.

File: server/http.go (hunk 3 above)
Comment (jespino) [reply server/http.go:253]: Yes, I have to investigate that, will be fixed soon :)

**Comment by larkox** [comment]
@jespino it seems there are still some issues with the test. Could you check them?

**Comment by jespino** [comment]
@larkox done

**Comment by larkox** [comment]
Thanks!
@DHaussermann heads up on this being ready for QA Review :smile:

**Comment by jespino** [comment]
@asaadmahmood If you consider this ok, can you approve the PR and remove the UX Review label?

**Pull request merged**